python/lsst/daf/butler/_butler.py

```diff
@@ -619,31 +619,34 @@ def _findDatasetRef(self, datasetRefOrType: Union[DatasetRef, DatasetType, str],
         # Go through the updated dataId and check the type in case someone is
         # using an alternate key.  We have already filtered out the compound
         # keys dimensions.record format.
-        # Will have to include kwds in this check
         not_dimensions = {}
-        # Use a list so we can adjust the dict safely in the loop
-        for dimensionName in list(newDataId):
-            value = newDataId[dimensionName]
-            try:
-                dimension = self.registry.dimensions[dimensionName]
-            except KeyError:
-                # This is not a real dimension
-                not_dimensions[dimensionName] = value
-                del newDataId[dimensionName]
-                continue
-
-            if not isinstance(value, dimension.primaryKey.getPythonType()):
-                for alternate in dimension.alternateKeys:
-                    if isinstance(value, alternate.getPythonType()):
-                        byRecord[dimensionName][alternate.name] = value
-                        del newDataId[dimensionName]
-                        log.debug("Converting dimension %s to %s.%s=%s",
-                                  dimensionName, dimensionName, alternate.name, value)
-                        break
-                else:
-                    log.warning("Type mismatch found for value '%r' provided for dimension %s. "
-                                "Could not find matching alternative (primary key has type %s).",
-                                value, dimensionName, dimension.primaryKey.getPythonType())
+
+        # Will need to look in the dataId and the keyword arguments
+        # and will remove them if they need to be fixed or are unrecognized.
+        for dataIdDict in (newDataId, kwds):
+            # Use a list so we can adjust the dict safely in the loop
+            for dimensionName in list(dataIdDict):
+                value = dataIdDict[dimensionName]
+                try:
+                    dimension = self.registry.dimensions[dimensionName]
+                except KeyError:
+                    # This is not a real dimension
+                    not_dimensions[dimensionName] = value
+                    del dataIdDict[dimensionName]
+                    continue
+
+                if not isinstance(value, dimension.primaryKey.getPythonType()):
```

Comment:
Good spot. It still worked but issued a warning but since `2` and `numpy.int64(2)` were still the "right" answer for the later check it made no difference. I now force integral types to an int here.